apis/python/src/tiledb/vector_search/index.py

```diff
@@ -686,6 +696,7 @@ def _query_additions(
         nthreads=8,
         timestamp=None,
         config=None,
+        distance_metric=vspy.DistanceMetric.L2,
```

Comment:
Do we need a default here? Can we remove it and have this be a required argument? Seems nice to avoid defaults for private methods to prevent errors where someone forgets to pass `distance_metric`.

Same with `create_metadata()`, `declare_vq_query_heap()`, etc - basically if we do pass an argument to the function, let's not have a default.

Reply:
I see this pattern in a lot of places with stuff that was there before - like group_exists and config in create_metadata - are we sure we want to change this only for distance_metric?

Reply:
Yep we should change those, but, either way, we don't have defaults for the other 5 arguments in `create_metadata()`, so we can just follow that pattern instead.